### ISSUE TITLE
CLI Build Documentation Issues

### ISSUE THREAD

**Issue by grady** [APEXCalculus/APEXCalculusPTX#186]
I'm just now trying to use new the new pretext CLI to build for the first time, starting from a fresh repository clone. I used pip to install pretextbook. I'm opening this to document any problems I have following the README during the process, as things aren't going 100% smoothly. 

- README says to edit Webwork URL in `project.ptx`, but no such option exists in that file. Should this refer to the `publication/publication.ptx` file instead?

- `prextext build html -d -w` is throwing a error that it can't find nonexistent file `source/main.ptx`.

```
(base) gradysw@janus:~/src/APEXCalculusUpstream$ pretext build html -d -w
No server name, 'server'.  Using default https://webwork-ptx.aimath.org
PTX: string parameters passed to extraction stylesheet: {}
PTX: XSL conversion of ~/src/APEXCalculusUpstream/source/main.ptx by ~/.local/lib/python3.7/site-packages/pretext/static/xsl/extract-pg.xsl
Traceback (most recent call last):
  File "~/.local/bin/pretext", line 10, in <module>
    sys.exit(main())
  File "~/anaconda3/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "~/anaconda3/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "~/anaconda3/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "~/anaconda3/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "~/anaconda3/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "~/.local/lib/python3.7/site-packages/pretext/cli.py", line 152, in build
    build.webwork(source, webwork_output, stringparams, server_params)
  File "~/.local/lib/python3.7/site-packages/pretext/build.py", line 172, in webwork
    core.webwork_to_xml(xml_source=ptxfile, pub_file=None, stringparams=params, abort_early=True, server_params=server_params, dest_dir=dest_dir)
  File "~/.local/lib/python3.7/site-packages/pretext/core.py", line 24, in webwork_to_xml
    ptxcore.webwork_to_xml(xml_source, pub_file, stringparams, abort_early, server_params, dest_dir)
  File "~/.local/lib/python3.7/site-packages/pretext/static/pretext/pretext.py", line 495, in webwork_to_xml
    xsltproc(extraction_xslt, xml_source, ww_filename, None, stringparams)
  File "~/.local/lib/python3.7/site-packages/pretext/static/pretext/pretext.py", line 1745, in xsltproc
    src_tree = ET.parse(xml, parser=huge_parser)
  File "src/lxml/etree.pyx", line 3467, in lxml.etree.parse
  File "src/lxml/parser.pxi", line 1839, in lxml.etree._parseDocument
  File "src/lxml/parser.pxi", line 1865, in lxml.etree._parseDocumentFromURL
  File "src/lxml/parser.pxi", line 1769, in lxml.etree._parseDocFromFile
  File "src/lxml/parser.pxi", line 1163, in lxml.etree._BaseParser._parseDocFromFile
  File "src/lxml/parser.pxi", line 601, in lxml.etree._ParserContext._handleParseResultDoc
  File "src/lxml/parser.pxi", line 711, in lxml.etree._handleParseResult
  File "src/lxml/parser.pxi", line 638, in lxml.etree._raiseParseError
OSError: Error reading file '~/src/APEXCalculusUpstream/source/main.ptx': failed to load external entity "~/src/APEXCalculusUpstream/source/main.ptx"
(base) gradysw@janus:~/src/APEXCalculusUpstream$ pretext
```

**Comment by grady** [comment]
OK, the previous error about main.ptx is because I was using python 3.7, and the pip package pretextbook was old there. I now notice that the requirements were for python=>3.10. 

Now with a new conda environment with more recent python and pretextbook, we also get new errors. The command `pretext build html -d -w` is obsolete, the help suggests `pretext generate -t html` or maybe `pretext build html -g` instead. The first of those is now running, we shall see if it does anything succesfully.

**Comment by grady** [comment]
OK, that was actually much less painful than I initially expected, given the immediate wall of error text on the first attempt. I have successfully built a document with the first 3 chapters of the book. Once things are working it seems to be a lot nicer than the old build method.

Some summary findings: 

- You don't seem to actually need python >=3.10. I seem to have had success with 3.8. But 3.7 does not work. So maybe the python requirement can be loosened to >=3.8?
- The webwork URL modification instruction in the README points you to the wrong file. The setting now appears to be in the publication file.
- The command in the README to generate images, etc, is obsolete in the pretextbook pip version. The command should now be `pretext generate` as stated in the previous comment.

**Comment by sean-fitzpatrick** [comment]
Thanks for pointing this out. The README needs to be updated to reflect some changes to PreTeXt that took place over the last month or so (mostly during the Michigan workshop). The build syntax for the CLI has changed, and Alex added the infrastructure needed to specify the WW server in the publication file instead of the project manifest.

**Comment by sean-fitzpatrick** [comment]
I think all of this is now addressed, once #187 is merged.
